### ISSUE TITLE
Change CategoryCodes to TermDefinitions

### ISSUE THREAD

**Issue by RichardWallis** [schemaorg/schemaorg#1775]
Following discussion in issue #1711 about the use of CategoryCodes (introduced into pending in #1255 from issue #894) there is a consensus that the terms should be renamed as follows:

- CategoryCode->TermDefinition
- CategoryCodeSet->TermDefinitionSet
- inCodeSet->inDefinedTermSet
- codeValue->termCode
- hasCategoryCode->hasTermDefinition

This issue created to carry out that work

**Comment by l00mi** [comment]
@RichardWallis Thank you for pushing this!

**Comment by philbarker** [comment]
@RichardWallis Thanks for the name change.

What about the changes to the definitions of DefinedTerm, DefinedTermSet and termCode?

DefinedTerm is currently 
>  a defined term

Suggest a more helpful definition of
> a word, name, acronym, phrase, etc. with a formal definition. Often used in the context of category or subject classification, glossaries or dictionaries, product or creative work types, etc. Use the name property for the term being defined, use termCode if the term has an alpha-numeric code allocated, use description to provide the definition of the term.

termCode is currently
> A short textual code that uniquely identifies the term.

Suggest that codes are only locally unique (and does textual include numbers?)
> A code that identifies this DefinedTerm within a DefinedTermSet

DefinedTermSet is currently
>A set of Defined Terms.

Suggest to help clarify what scope is
> A set of defined terms for example a set of categories or a classification scheme, a glossary, dictionary or enumeration.

**Comment by RichardWallis** [comment]
Thanks @philbarker for pointing out that I forgot to change the descriptions - I will get on to it asap.

**Comment by philbarker** [comment]
@RichardWallis I have made changes locally, would it help if I sent a pull request?

**Comment by RichardWallis** [comment]
It would make sense for me to update the current PR (#1776), instead of creating yet another one, so I'll do that

**Comment by RichardWallis** [comment]
Updated descriptions now part of PR (#1776)

**Comment by RichardWallis** [comment]
Implemented in PR #1776

**Comment by jaygray0919** [comment]
@RichardWallis i'm late to the party here, but would just like to confirm these is active:
```
https://schema.org/DefinedTerm
https://schema.org/CategoryCode
```
and that all of the above discussion is related to the current specification for `@DefinedTerm` and `@CategoryCode`.
We use those two types and their properties extensively; was worried that a change might be coming.

**Comment by RichardWallis** [comment]
@jaygray0919 no worries, both of these terms are active.

This issue has only been closed now, as a result of an exercise (#2573) to identify open issues that should have been previously closed.
